core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -197,6 +207,7 @@ protected final void doStart() throws Exception {
     HandlerCollection wsHandlers = new HandlerCollection();
     for (Application<?> app : applications) {
       attachMetricsListener(app.getMetrics(), app.getMetricsTags());
+      attachNetworkTrafficRateLimitListener();
```

Comment:
Does the order you add them here match the order that the listeners run in?  We want the metrics listener to fire before the rate limiting one, so that we get metrics about everything trying to connect.

I the right ordering matches what you've done, but I wanted to ask if you'd checked.

Reply:
Yes, underneath it is a list of listeners so those are added first will be run first.